### ISSUE TITLE
Problem upgrading Node to Latest Current Version v8.5.0, npm 5.3.0

### ISSUE THREAD

**Issue by dennisC61** [nodejs/help#841]
I started with 
Node.js Version 6.10.3 
npm 3.10.10

I needed to upgrade my versions to Latest Current Version;
Node.js Version 8.5.0
npm 5.3.0

My OS is Linux Cent OS 7

I ran these two lines at my root directory

```console
[root@boot /] # curl --silent --location https://rpm.nodesource.com/setup_8.x | sudo bash -
[root@boot /] # sudo yum -y install nodejs
```

The result was a slight upgrade to Node.js 6.10.3->6.11.1
and npm version stayed at 3.10.10.

I was expecting an upgrade to Node.js 8.5.0
and npm version 5.3.0

Is there any additional syntax that I can run on the command line to upgrade to the 
Latest Current Version;
Node.js Version 8.5.0
npm 5.3.0

Thank You
Dennis

Here is the command\response dump

```console
[root@boot /] # curl --silent --location https://rpm.nodesource.com/setup_8.x | sudo bash -
[root@boot /] # sudo yum -y install nodejs
Loaded plugins: fastestmirror, langpacks
base                                                                                                                    | 3.6 kB  00:00:00     
epel/x86_64/metalink                                                                                                    |  13 kB  00:00:00     
epel                                                                                                                    | 4.3 kB  00:00:00     
extras                                                                                                                  | 3.4 kB  00:00:00     
updates                                                                                                                 | 3.4 kB  00:00:00     
(1/4): extras/7/x86_64/primary_db                                                                                       | 102 kB  00:00:00     
(2/4): epel/x86_64/updateinfo                                                                                           | 830 kB  00:00:00     
(3/4): updates/7/x86_64/primary_db                                                                                      | 2.9 MB  00:00:01     
(4/4): epel/x86_64/primary_db                                                                                           | 4.8 MB  00:00:02     
Loading mirror speeds from cached hostfile
* base: mirror.pac-12.org
* epel: mirror.sjc02.svwh.net
* extras: centos.sonn.com
* updates: centos.sonn.com
Resolving Dependencies
--> Running transaction check
---> Package nodejs.x86_64 1:6.10.3-1.el7 will be updated
--> Processing Dependency: nodejs = 1:6.10.3-1.el7 for package: 1:npm-3.10.10-1.6.10.3.1.el7.x86_64
---> Package nodejs.x86_64 1:6.11.1-1.el7 will be an update
--> Running transaction check
---> Package npm.x86_64 1:3.10.10-1.6.10.3.1.el7 will be updated
---> Package npm.x86_64 1:3.10.10-1.6.11.1.1.el7 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

===============================================================================================================================================
Package                       Arch                          Version                                         Repository                   Size
===============================================================================================================================================
Updating:
nodejs                        x86_64                        1:6.11.1-1.el7                                  epel                        4.5 M
Updating for dependencies:
npm                           x86_64                        1:3.10.10-1.6.11.1.1.el7                        epel                        2.5 M

Transaction Summary
===============================================================================================================================================
Upgrade  1 Package (+1 Dependent package)

Total download size: 7.0 M
Downloading packages:
No Presto metadata available for epel
(1/2): npm-3.10.10-1.6.11.1.1.el7.x86_64.rpm                                                                            | 2.5 MB  00:00:01     
(2/2): nodejs-6.11.1-1.el7.x86_64.rpm                                                                                   | 4.5 MB  00:00:01     
-----------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                          3.6 MB/s | 7.0 MB  00:00:01     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : 1:nodejs-6.11.1-1.el7.x86_64                                                                                                1/4 
  Updating   : 1:npm-3.10.10-1.6.11.1.1.el7.x86_64                                                                                         2/4 
  Cleanup    : 1:nodejs-6.10.3-1.el7.x86_64                                                                                                3/4 
  Cleanup    : 1:npm-3.10.10-1.6.10.3.1.el7.x86_64                                                                                         4/4 
  Verifying  : 1:npm-3.10.10-1.6.11.1.1.el7.x86_64                                                                                         1/4 
  Verifying  : 1:nodejs-6.11.1-1.el7.x86_64                                                                                                2/4 
  Verifying  : 1:npm-3.10.10-1.6.10.3.1.el7.x86_64                                                                                         3/4 
  Verifying  : 1:nodejs-6.10.3-1.el7.x86_64                                                                                                4/4 

Updated:
  nodejs.x86_64 1:6.11.1-1.el7                                                                                                                 

Dependency Updated:
  npm.x86_64 1:3.10.10-1.6.11.1.1.el7                                                                                                          

Complete!
[root@boot /] # node -v
v6.11.1
[root@boot /] # npm -v
3.10.10
[root@boot /] #
```

**Comment by kasicka** [comment]
yum installed nodejs available from epel. Did you try ```yum clean all``` and is nodesource repo between your enabled repositories?

**Comment by dennisC61** [comment]
I haven't tried yum clean all yet, but I will.
I don't quite understand what is meant by, 'is nodesource repo between your enabled repositories?'
Thanks for the response!

**Comment by dennisC61** [comment]
I did try yum clean all, but got the same result, versions unchanged from node 6.11.1, npm 3.10.10.

**Comment by kasicka** [comment]
Does ```yum install https://rpm.nodesource.com/pub_8.x/el/7/x86_64/nodesource-release-el7-1.noarch.rpm && yum install nodejs``` help?
```yum repolist``` shows if you have nodesource repo enabled

**Comment by dennisC61** [comment]
I tried the yum install command and it says it can't open the url

[root@boot /]# yum install https://rpm.nodesource.com/pub_8.x/el/7/x86_64/nodesource-release-el7-1.noarch.rpm
Loaded plugins: fastestmirror, langpacks
Cannot open: https://rpm.nodesource.com/pub_8.x/el/7/x86_64/nodesource-release-el7-1.noarch.rpm. Skipping.
Error: Nothing to do
[root@boot /]#

**Comment by kasicka** [comment]
Works fine for me.

**Comment by dennisC61** [comment]
I ran yum repolist all and didn't see nodesource listed as one of the yum repositories on my machine.

I'll try typing in your yum install command and see what happens. Before I did a copy\paste.

**Comment by dennisC61** [comment]
I typed in the yum install command and got the same response.

[root@boot /]# yum install https://rpm.nodesource.com/pub_8.x/el/7/x86_64/nodesource-release-el7-1.noarch.rpm
Loaded plugins: fastestmirror, langpacks
Cannot open: https://rpm.nodesource.com/pub_8.x/el/7/x86_64/nodesource-release-el7-1.noarch.rpm. Skipping.
Error: Nothing to do
[root@boot /]#

**Comment by dennisC61** [comment]
Do I need to install and enable the nodesource repository on my machine first?
I'm new to yum, just read about this command
yum-config-manager --add-repo repository_url

If yes, then what is the url to the nodesource repository?

**Comment by gibfahn** [comment]
@dennisC61 instructions are https://nodejs.org/en/download/package-manager/

If it is missing commands you need, please raise an issue at https://github.com/nodesource/distributions.

**Comment by dennisC61** [comment]
Zuzana

I’m still working on updating my node and npm versions.

When I list all repositories in my CENT OS 7 system, I don’t see a nodesource repository.

Do I need to run a specific yum command to first create the nodesource repository?
Yum –enablerepo =

Below is a listing of all of my repositories.

Dennis

[huskiesDev@boot /]$ yum repolist all
Loaded plugins: fastestmirror, langpacks
Determining fastest mirrors
* base: mirror.nodesdirect.com
* epel: mirrors.cat.pdx.edu
* extras: mirror.atlanticmetro.net
* updates: mirror.steadfast.net
repo id                       repo name                          status
C7.0.1406-base/x86_64         CentOS-7.0.1406 - Base             disabled
C7.0.1406-centosplus/x86_64   CentOS-7.0.1406 - CentOSPlus       disabled
C7.0.1406-extras/x86_64       CentOS-7.0.1406 - Extras           disabled
C7.0.1406-fasttrack/x86_64    CentOS-7.0.1406 - CentOSPlus       disabled
C7.0.1406-updates/x86_64      CentOS-7.0.1406 - Updates          disabled
C7.1.1503-base/x86_64         CentOS-7.1.1503 - Base             disabled
C7.1.1503-centosplus/x86_64   CentOS-7.1.1503 - CentOSPlus       disabled
C7.1.1503-extras/x86_64       CentOS-7.1.1503 - Extras           disabled
C7.1.1503-fasttrack/x86_64    CentOS-7.1.1503 - CentOSPlus       disabled
C7.1.1503-updates/x86_64      CentOS-7.1.1503 - Updates          disabled
C7.2.1511-base/x86_64         CentOS-7.2.1511 - Base             disabled
C7.2.1511-centosplus/x86_64   CentOS-7.2.1511 - CentOSPlus       disabled
C7.2.1511-extras/x86_64       CentOS-7.2.1511 - Extras           disabled
C7.2.1511-fasttrack/x86_64    CentOS-7.2.1511 - CentOSPlus       disabled
C7.2.1511-updates/x86_64      CentOS-7.2.1511 - Updates          disabled
!base/7/x86_64                CentOS-7 - Base                    enabled:  9,591
base-debuginfo/x86_64         CentOS-7 - Debuginfo               disabled
base-source/7                 CentOS-7 - Base Sources            disabled
c7-media                      CentOS-7 - Media                   disabled
centosplus/7/x86_64           CentOS-7 - Plus                    disabled
centosplus-source/7           CentOS-7 - Plus Sources            disabled
cr/7/x86_64                   CentOS-7 - cr                      disabled
*!epel/x86_64                 Extra Packages for Enterprise Linu enabled: 11,953
epel-debuginfo/x86_64         Extra Packages for Enterprise Linu disabled
epel-source/x86_64            Extra Packages for Enterprise Linu disabled
epel-testing/x86_64           Extra Packages for Enterprise Linu disabled
epel-testing-debuginfo/x86_64 Extra Packages for Enterprise Linu disabled
epel-testing-source/x86_64    Extra Packages for Enterprise Linu disabled
!extras/7/x86_64              CentOS-7 - Extras                  enabled:    223
extras-source/7               CentOS-7 - Extras Sources          disabled
fasttrack/7/x86_64            CentOS-7 - fasttrack               disabled
google-chrome                 google-chrome                      disabled
!updates/7/x86_64             CentOS-7 - Updates                 enabled:    719
updates-source/7              CentOS-7 - Updates Sources         disabled
repolist: 22,486
[huskiesDev@boot /]$

**Comment by kinsomicrote** [comment]
Have you tried using a Node version manager?

**Comment by ORESoftware** [comment]
https://github.com/creationix/nvm

very recommended

**Comment by kinsomicrote** [comment]
Here is an alternative

https://github.com/nodenv/nodenv

**Comment by dennisC61** [comment]
Kingsley

Thanks for the suggestion.

I came to realize that I was having internet connectivity issues through a Virtual Machine running the Linux Cent OS 7 and the
the connectivity issues were causing command line accesses to url’s on GitHub to fail.

I did develop a work around in which I downloaded the .rpm package file nodejs-8.5.0-1nodesource.x86_64.rpm on windows, copy into the VM, and installed with the rpm tool. Testing so far indicates that this is a viable work around.

Thanks to everyone who responded to my posts.

Dennis
